core/arch/arm/sm/pm_a32.S

```diff
@@ -147,6 +147,20 @@ UNWIND(	.fnend)
 END_FUNC sm_pm_cpu_resume
 
 /*
+ * The following will be located in text section whose attribute is
+ * marked as readonly, but we only need to read here
+ * _suspend_sp stores the offset between thread_core_local to _suspend_sp.
+ * _core_pos stores the offset between get_core_pos to _core_pos.
+ */
+.align 2
+.extern thread_core_local
+_suspend_sp:
+	.long	thread_core_local - .
+.extern get_core_pos
+_core_pos:
+	.long	get_core_pos - .
+
+/*
```

Comment:
Why move this block here? Just cosmetic?

Reply:
It's to avoid an assembler error, apparantly the `adr` instruction can only be performed on symbols in the same section.